source/common/config/delta_subscription_impl.cc

```diff
@@ -39,6 +39,12 @@ void DeltaSubscriptionImpl::updateResourceInterest(
   stats_.update_attempt_.inc();
 }
 
+void DeltaSubscriptionImpl::addResourceInterest(const std::set<std::string>& add_these_names) {
+  watch_ = context_->addOrUpdateWatch(type_url_, watch_, add_these_names, *this,
```

Comment:
I don't think `GrpcMux::addOrUpdateWatch` does what you are needing it to do. You are trying to use it as "add TO the existing watch", but what it's actually going to do is just overwrite with add_these_names. The "add" part of "addOrUpdateWatch" means "add a watch to the map if there was none" (i.e. the watch argument is nullptr).

So, looks like this function was named entirely from the point of view of someone (i.e. me) implementing GrpcMux and WatchMap, rather than what would make sense for an external user. My apologies! Maybe it should just be called `updateWatch`.

Anyways, I think what you need to do is add a function to WatchMap. I think what you want is to give `WatchMap::updateWatchInterest` a sibling, maybe `WatchMap::addToWatchInterest`, that basically does the exact same as what updateWatchInterest does, except `newly_added_to_watch` is just the input argument, and `newly_removed_from_watch` is the empty set.